### PR TITLE
Sidebar Layout

### LINKED ISSUE #755: First / Last Played date format shoud honour system date format
The first / last played date format appears to be hard coded to mm/dd/yy (American) when using English language whereas my system is dd/mm/yy (British).

See attached screenshot

Screenshot_20211122_181057

### PR DESCRIPTION
**Pull request by flavioislima**
This changes Heroic UI to use a Sidebar instead of a Navbar. 
Changes were made on the Game page as well.
Several other components were refactored and a few fixes as well.

Fix #755 
Implement #431 
Implement #433 

image
image
image

-------------------------------------------------------------
Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [x] Created / Updated Tests (If necessary)
- [x] Created / Updated documentation (If necessary)

### CODE REVIEW

**Review by dragonDScript — COMMENTED** [review]
Please don't create such big PRs, this pr includes stuff not related to the sidebar.

**Comment by flavioislima** [comment]
> Please don't create such big PRs, this pr includes stuff not related to the sidebar.

Yes, you're right. There are a few fixes and we also changed the GamePage because otherwise it would be broken.
But ignore the JSON files and focus on the react ones.

**Review by adityaruplaha — COMMENTED** [review]
I'm no UI expert, so I've not looked too deep into the CSS. This PR doesn't touch the backend much, so I don't have much to say. But some minor things drew my eye.

File: package.json
Comment (flavioislima) [reply package.json:190]: Man don't bother about trailing newlines, focus on the code.
I think I probably removed some boring eslint rules that were making us lose time because we saved the code and then got an error because of some rule that doesn't do any good. It's just styling. We should focus on rules that could catch errors or make our lives easier. 👍🏽

File: src/App.tsx
Comment (flavioislima) [reply src/App.tsx:8]: what do you mean?

File: src/screens/Game/GamePage/index.tsx
Comment (flavioislima) [reply src/screens/Game/GamePage/index.tsx:231]: Good catch, forgot to remove those 👍🏽

File: src/screens/Game/TimeContainer/index.tsx
Comment (flavioislima) [reply src/screens/Game/TimeContainer/index.tsx:45]: To fix #755.
This function needs to receive a locale or undefined, yes, weird. But this is JS 😅 
So before that we were using the user language to format the date but this makes no sense since you might live in a country that speaks English but the date format is not from US. So with this change we will use the system format.

File: src/App.tsx
Comment (adityaruplaha) [reply src/App.tsx:8]: Like are we using './...' or 'src/'.
It's just styling, don't worry too much about it.

**Review by adityaruplaha — APPROVED** [review]
LGTM! If tests pass, all should be fine.

File: src/screens/Library/components/GameCard/index.tsx
Comment (flavioislima) [reply src/screens/Library/components/GameCard/index.tsx:10]: I think at some point we lost the plugin or configuration to fix the styling. I think when we removed prettier at some point so now some things are not getting autofixed. 
At least for me, it fixes the indentation and other things but not everything.

File: src/components/UI/InfoBox/index.css
Comment (dragonDScript) [reply src/components/UI/InfoBox/index.css:28]: No, we should try to make the code as clean as possible. You can commit changes within the github UI, so it shouldn't be a problem.

**Comment by adityaruplaha** [comment]
I think we shouldn't worry too much about styling in this PR. Just the code.
Once its merged, we can autofix almost everything using the right plugins. Plus its not horribly unreadable (yet), so it can wait.

**Comment by flavioislima** [comment]
> I think we shouldn't worry too much about styling in this PR. Just the code. Once its merged, we can autofix almost everything using the right plugins. Plus its not horribly unreadable (yet), so it can wait.

Yes, I'll reconsider prettier or another plugin that can make this automatically

File: src/App.tsx
Comment (flavioislima) [reply src/App.tsx:8]: We are using absolute paths for quite some time now.

**Pull request merged**
